### DISCUSSION TITLE
Implicit breaks on switch cases

### DISCUSSION THREAD

**Discussion by JustNrik** [dotnet/csharplang#5721] [Language Ideas]
I'm completely aware about why c# has disabled implicit, fallthrough, this isn't about that just in case.

Currently, the only implicit fallthrough we have is through consecutive empty cases, otherwise, you **must** use `goto`. So the question is, if we can't fallthrough unless the case is empty, and the only other way we can fallthrough is through `goto`, then why do we have to use `break` on every other case? It really makes no sense.

I may think it's for readability or explicitness, but it's not. It kills readability and it's completely unnecessary. It's verbose. My suggestion is, make `break` implicit unless the compiler determines it's necessary (can't really think of a case where it is, but whatever). 

Making `break` implicit will greatly improve the readability as it reduces the noisy of having `break` on every case.

**Comment by theunrepentantgeek** [comment]
In most cases, two statements with different syntax should have different meaning.

Consider what a reader might thing when reading a switch statement like this:

``` csharp
case Foo:
    DoTheFooThing();
    break;
case Bar:
    DoTheBarThing();
case Baz:
    DoTheBazThing();
```

One of the branches has a **break** statement, the other does not. 

This _strongly_ implies a difference in behaviour.

Even if the branches were in completely different switch statements, the difference in syntax would lead readers to assume a difference exists, especially if they have experience in any of the myriad of { curly-brace } languages that use _implicit fall through_.

> I may think it's for readability or explicitness, but it's not. It kills readability and it's completely unnecessary. It's verbose.

Harking back to the original decision-making process of C#, I recall articles from around the year 2000 talking about how the explicit break was specifically for clarity and to avoid confusion. 

If we were designing a C# style language from scratch here in 2022, switch statements would be different. 

But, we're not.
